src/Sylius/Bundle/CoreBundle/DataFixtures/Factory/AddressFactory.php

```diff
@@ -0,0 +1,155 @@
+ * @method static AddressInterface[]|Proxy[] randomSet(int $number, array $attributes = [])
+ * @method static AddressInterface[]|Proxy[] randomRange(int $min, int $max, array $attributes = [])
+ * @method AddressInterface|Proxy create(array|callable $attributes = [])
+ */
+final class AddressFactory extends ModelFactory implements AddressFactoryInterface
+{
+    public function __construct(
+        private FactoryInterface $addressFactory,
+        private CountryFactoryInterface $countryFactory,
+        private ShopUserFactoryInterface $shopUserFactory
+    ) {
+        parent::__construct();
+    }
+
+    public function withFirstName(string $firstName): self
+    {
+        return $this->addState(['first_name' => $firstName]);
+    }
+
+    public function withLastName(string $lastName): self
+    {
+        return $this->addState(['last_name' => $lastName]);
+    }
+
+    public function withPhoneNumber(string $phoneNumber): self
+    {
+        return $this->addState(['phone_number' => $phoneNumber]);
+    }
+
+    public function withCompany(string $company): self
+    {
+        return $this->addState(['company' => $company]);
+    }
+
+    public function withStreet(string $street): self
+    {
+        return $this->addState(['street' => $street]);
+    }
+
+    public function withCity(string $city): self
+    {
+        return $this->addState(['city' => $city]);
+    }
+
+    public function withPostcode(string $postcode): self
+    {
+        return $this->addState(['postcode' => $postcode]);
+    }
+
+    public function withCountryCode(string $countryCode): self
+    {
+        return $this->addState(['country_code' => $countryCode]);
+    }
+
+    public function withProvinceName(string $provinceName): self
+    {
+        return $this->addState(['province_name' => $provinceName]);
+    }
+
+    public function withProvinceCode(string $provinceCode): self
+    {
+        return $this->addState(['province_code' => $provinceCode]);
+    }
+
+    public function withCustomer(Proxy|CustomerInterface|string $customer): self
+    {
+        return $this->addState(function () use ($customer): array {
+            if (is_string($customer)) {
+                $customer = $this->shopUserFactory::randomOrCreate(['email' => $customer])->getCustomer();
+            }
+
+            return ['customer' => $customer];
+        });
+    }
+
+    protected function getDefaults(): array
+    {
+        return [
+            'first_name' => self::faker()->firstName(),
+            'last_name' => self::faker()->lastName(),
+            'phone_number' => self::faker()->boolean() ? self::faker()->phoneNumber() : null,
+            'company' => self::faker()->boolean() ? self::faker()->company() : null,
+            'street' => self::faker()->streetAddress(),
+            'city' => self::faker()->city(),
+            'postcode' => self::faker()->postcode(),
+            'country_code' => $this->countryFactory::randomOrCreate()->getCode(),
```

Comment:
Why are we calling a static method from an object ?
Wouldn't it be easier to just not inject the dependency and call the Static from class name ?

Reply:
We do not use parameters for the class names of the factories. When using your own factory service, you just use a service alias.

```yaml
services:
   Sylius\Bundle\CoreBundle\DataFixtures\Factory\CountryFactoryInterface: App\DataFixtures\Factor\CountryFactory
```

For what you suggest, you have to change a parameter for the factory class like this:
```yaml
parameters:
    sylius.data_fixtures.factory.country.class: App\DataFixtures\Factor\CountryFactory
```

But I think it's the legacy way and not Symfony-like. I didn't choose this type of DI configuration.